File: src/fiefapp.svelte/src/models/resident.ts
import type { IPerson } from './person';

interface IResident {
  information: string;
}

export class Resident implements IResident, IPerson {
  id: string;
  name: string;
  age: number;
  resources: number;
  loyalty: number;
  skill: number;
  information: string;

  constructor() {
    this.id = '';
    this.name = '';
    this.age = 0;
    this.resources = 0;
    this.skill = 0;
    this.loyalty = 0;
    this.information = '';
  }
}

File: src/fiefapp.entities/Resident.cs
using fiefapp.entities.BaseClasses;
using fiefapp.entities.Interfaces;
using System;
using System.Collections.Generic;

namespace fiefapp.entities
{
    public class Resident : Person, IBaseEntity, IFiefForeinKey
    {
        public override Guid Id { get; set; }
        public Guid FiefId { get; set; }
        public override string Name { get; set; }
        public override int Age { get; set; }
        public override int Resources { get; set; }
        public override int Loyalty { get; set; }
        public override int Skill { get; set; }
        public override ICollection<string> Notes { get; set; }
    }
}
